File: iosApp/iosApp/Views/Schedule/List/MilestoneListItem.swift
//
//  MilestoneListItem.swift
//  iosApp
//
//  Copyright © 2023 Ludwig Boltzmann Institute for
//  Digital Health and Prevention - A research institute
//  of the Ludwig Boltzmann Gesellschaft,
//  Oesterreichische Vereinigung zur Foerderung
//  der wissenschaftlichen Forschung
//  Licensed under the Apache 2.0 license with Commons Clause
//  (see https://www.apache.org/licenses/LICENSE-2.0 and
//  https://commonsclause.com/).
//

import shared
import SwiftUI

struct MilestoneSection: View {
    var milestones: [MilestoneEntity]

    var body: some View {
        VStack(alignment: .leading) {
            BasicText(text: String(localized: "milestones"), color: Color.more.primaryDark)
                .font(Font.more.headline)
            Divider()
            ForEach(milestones, id: \.participantMilestoneId) { milestone in
                MilestoneListItem(milestone: milestone)
                Divider()
            }
        }
        .padding(.bottom)
    }
}

struct MilestoneListItem: View {
    var milestone: MilestoneEntity

    var body: some View {
        HStack {
            Image(systemName: "rosette")
                .foregroundColor(Color.more.primary)
            BasicText(text: milestone.name)
                .font(Font.more.headline)
                .foregroundColor(Color.more.primary)
            Spacer()
            Text(milestone.dateTime.toDateString(dateFormat: "dd.MM.yyyy"))
                .foregroundColor(Color.more.secondary)
        }
        .padding(.vertical, 8)
    }
}

File: androidApp/src/main/java/io/redlink/more/app/android/activities/dashboard/schedule/list/MilestoneListItem.kt
/*
 * Copyright LBI-DHP and/or licensed to LBI-DHP under one or more
 * contributor license agreements (LBI-DHP: Ludwig Boltzmann Institute
 * for Digital Health and Prevention -- A research institute of the
 * Ludwig Boltzmann Gesellschaft, Österreichische Vereinigung zur
 * Förderung der wissenschaftlichen Forschung).
 * Licensed under the Apache 2.0 license with Commons Clause
 * (see https://www.apache.org/licenses/LICENSE-2.0 and
 * https://commonsclause.com/).
 */
package io.redlink.more.app.android.activities.dashboard.schedule.list

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Icon
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import io.redlink.more.app.android.R
import io.redlink.more.app.android.extensions.formattedString
import io.redlink.more.app.android.extensions.getStringResource
import io.redlink.more.app.android.extensions.jvmLocalDate
import io.redlink.more.app.android.shared_composables.BasicText
import io.redlink.more.app.android.shared_composables.Heading
import io.redlink.more.app.android.shared_composables.MoreDivider
import io.redlink.more.app.android.shared_composables.SmallTitle
import io.redlink.more.app.android.theme.MoreColors
import io.redlink.more.database.entities.MilestoneEntity

@Composable
fun MilestoneSection(milestones: List<MilestoneEntity>, modifier: Modifier = Modifier) {
    Column(modifier = modifier.fillMaxWidth()) {
        Heading(text = getStringResource(R.string.more_milestone_section_title))
        milestones.forEach { milestone ->
            MoreDivider(Modifier.fillMaxWidth())
            MilestoneListItem(milestone)
        }
        Spacer(modifier = Modifier.height(30.dp))
    }
}

@Composable
fun MilestoneListItem(milestone: MilestoneEntity) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 16.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(
                Icons.Default.EmojiEvents,
                contentDescription = null,
                tint = MoreColors.Primary,
                modifier = Modifier.padding(end = 8.dp)
            )
            SmallTitle(text = milestone.name, color = MoreColors.Primary)
        }
        BasicText(
            text = milestone.dateTime.jvmLocalDate().formattedString(),
            color = MoreColors.Secondary
        )
    }
}
